### PR TITLE
fix DOMTokenList

### LINKED ISSUE #25128: Emptying a classList leaves getAttribute("class") as an empty string where WPT expects null
WPT dom/nodes/Element-classlist.html fails several classList.remove test cases this way.

### PR DESCRIPTION
**Pull request by pshaughn**
I fixed the bug described in 25128, and also the other cases in the same test (order of elements after replace).

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #25128 

- [X] There are tests for these changes

### CODE REVIEW

File: components/script/dom/domtokenlist.rs [hunk 1]
```diff
@@ -52,6 +52,16 @@ impl DOMTokenList {
             slice => Ok(Atom::from(slice)),
         }
     }
+
+    fn perform_update_steps(&self, atoms: Vec<Atom>) {
```
Comment (jdm) [inline components/script/dom/domtokenlist.rs]: Add a `// https://dom.spec.whatwg.org/#concept-dtl-update` comment please.

**Review by jdm — APPROVED** [review]

**Comment by jdm** [comment]
@bors-servo retry
* #24726

**Comment by CYBAI** [comment]
@bors-servo retry

- #24726

**Pull request merged**